src/Authenticator/TokenAuthenticator.php

```diff
@@ -44,12 +44,12 @@ class TokenAuthenticator extends AbstractAuthenticator implements StatelessInter
     protected function getToken(ServerRequestInterface $request)
     {
         $token = $this->getTokenFromHeader($request, $this->getConfig('header'));
-        if (empty($token)) {
+        if ($token === null) {
```

Comment:
Are we sure this can never be empty string?

Reply:
If it is empty string it will fail more below. Aka the credentials where given but did not match anything (empty username can be ok imho, basically)